test/integration/targets/ios_bgp/tests/cli/basic.yaml

```diff
@@ -5,390 +5,398 @@
+                masklen: 27
+
+          - afi: ipv4
+            safi: unicast
+            networks:
+              - prefix: 198.51.100.64
+                masklen: 28
+    register: result
+
+  - assert:
+      that:
+        - 'result.changed == true'
+        - '"router bgp 64496" in result.commands'
+        - '"address-family ipv4 multicast" in result.commands'
+        - '"network 198.51.100.80 mask 255.255.255.240" in result.commands'
+        - '"network 203.0.113.192 mask 255.255.255.224" in result.commands'
+        - '"no network 198.51.100.48 mask 255.255.255.240 route-map RMAP_1" in result.commands'
+        - '"no network 203.0.113.160 mask 255.255.255.224 route-map RMAP_2" in result.commands'
+        - '"exit-address-family" in result.commands'
+
+  - name: Configure networks under address family with operation replace (idempotent)
+    ios_bgp: *af_net_rplc
+    register: result
+
+  - assert:
+      that:
+        - 'result.changed == false'
+
+  - name: Configure redistribute information under address family mode
+    ios_bgp: &af_rdr
+      operation: merge
+      config:
+        bgp_as: 64496
+        address_family:
+          - afi: ipv4
+            safi: multicast
+            redistribute:
+              - protocol: ospf
+                id: 112
+                metric: 64
+
+              - protocol: eigrp
+                id: 233
+                metric: 256
+    register: result
+
+  - assert:
+      that:
+        - 'result.changed == true'
+        - "'router bgp 64496' in result.commands"
+        - "'address-family ipv4 multicast' in result.commands"
+        - "'redistribute ospf 112 metric 64' in result.commands"
+        - "'redistribute eigrp 233 metric 256' in result.commands"
+        - "'exit-address-family' in result.commands"
+
+  - name: Configure redistribute information under address family mode (idempotent)
+    ios_bgp: *af_rdr
+    register: result
+
+  - assert:
+      that:
+        - 'result.changed == false'
+
+  - name: Get the IOS version
+    ios_facts:
+      gather_subset: all
+
+  - name: Configure redistribute information under address family mode with operation replace
+    ios_bgp: &af_rdr_rplc
+      operation: replace
+      config:
+        bgp_as: 64496
+        address_family:
+          - afi: ipv4
+            safi: multicast
+            redistribute:
+              - protocol: ospf
+                id: 112
+                metric: 64
+    register: result
+
+  - assert:
+      that:
+        - 'result.changed == true'
+        - "'router bgp 64496' in result.commands"
+        - "'address-family ipv4 multicast' in result.commands"
+        - "'no redistribute eigrp 233' in result.commands"
+        - "'exit-address-family' in result.commands"
+
+  - name: Configure redistribute information under address family mode with operation replace (idempotent)
+    ios_bgp: *af_rdr_rplc
+    register: result
+    when: ansible_net_version != "15.6(2)T"
```

Comment:
Lets move this 1 task down, as we can still do the idempotent task, just not validate the results.